clippy_lints/src/copies.rs

```diff
@@ -365,6 +371,38 @@ fn eq_stmts(
         .all(|b| get_stmt(b).map_or(false, |s| eq.eq_stmt(s, stmt)))
 }
 
+fn contains_acceptable_macro(cx: &LateContext<'_>, block: &Block<'_>) -> bool {
+    for stmt in block.stmts {
```

Comment:
Could be `block.stmts.iter().any(|s| matches!(s.kind, StmtKind::Semi(e) if acceptable_macro(cx, e)))`

It also seems unnecessary to check all statements for the macro. `todo` isn't usually followed by other things in the same block.